lib/src/main/java/org/esbtools/eventhandler/Promise.java

```diff
@@ -24,22 +24,24 @@
  * An asynchronous capture of potentially yet-to-come responses.
  *
  * <p>Allows code to build a {@link Future} object which will complete at some point in the future
- * with results from the {@link ResponsesHandler} passed to {@link #then(ResponsesHandler)}.
+ * with results from the {@link PromiseHandler} passed to {@link #then(PromiseHandler)}.
  *
  * @see Requester
  *
  * @param <T> The type of requests
- * @param <U> The type of responses
+ * @param <T> The type of responses
  */
-public interface ResponsePromise<T, U> {
+public interface Promise<T> extends Future<T> {
```

Comment:
> I'm not sure what definitions of promise and future you're drawing from. Different languages and libraries use different definitions of "promise" and "future"
I'm going specifically off of java.util.Future which according to javadocs "represents the result of an asynchronous computation"; this does not imply anything about what is done once the future is resolved (i.e. the future itself does not have anything to do with callbacks).

When referring to promises, since Java doesn't have any promises implementations, I think it's only really sane to talk about promises in Javascript (though I'm open to talking about promises from other languages), so I'm referring to the promises standard in ECMAScript 2015.

> Ultimately what I'm saying is the terms don't matter so much.

I disagree. Language is important for clarity, especially given that others may need to maintain this in the future (no pun intended, but haha).

> What constitutes "information on how to resolve itself"? In what way is blocking to wait for a future result "resolving itself?"

I suppose I'm talking less about this interface, and more about how this interface is used in the overall solution. So, we can discuss this under `LazyPromise` if you'd like; specifically the implementation of `get`.

> Guava also has this same construct via `ListenableFuture`.

I would call out that the construct is similar conceptually, but not the same. I'm actually not a huge fan of `ListenableFuture` either, but it is better than this. What I think would actually be more appropriate is a separate class which is composed of a future (read has-a vs. is-a), and otherwise has a similar interface to `ListenableFuture` (I'd call this class `FutureCallbackWrapper` or something similar). This would be better because then you can simply use vanilla `Future` with callbacks with not much extra code (I'm actually surprised that implementations don't have constructors which accept futures...). In the end though, because `ListenableFuture` already exists and is in larger use, I would be okay with using it.